### DISCUSSION TITLE
Does "on_all_tables_in_schema" include dynamic tables?

### DISCUSSION THREAD

**Discussion by pslking** [Snowflake-Labs/terraform-provider-snowflake#3196] [Q&A] [answered]
Hi there. I'm trying to share all dynamic tables within a specific schema. I'm using the `snowflake_grant_privileges_to_share` resource and pointing `on_all_tables_in_schema` to the schema. This schema does not have any actual TABLES, just DYNAMIC TABLES, and I'm not seeing those in the share. Does this option include dynamic tables, or will I need to grant those to the share individually?

**Comment by sfc-gh-asawicki** [comment, marked as answer]
Hey @pslking. Dynamic tables were added to https://docs.snowflake.com/en/sql-reference/sql/grant-privilege-share after this resource was created. Currently, using `on_all_tables_in_schema` results in running this part `ALL TABLES IN SCHEMA <schema_name> }`.

Please create a feature request to support these options:
```
      | { EXTERNAL TABLE <name> | ALL EXTERNAL TABLES IN SCHEMA <schema_name> }
      | { ICEBERG TABLE <name> | ALL ICEBERG TABLES IN SCHEMA <schema_name> }
      | { DYNAMIC TABLE <name> | ALL DYNAMIC TABLES IN SCHEMA <schema_name> }
```

**Reply by pslking** [reply]
Feature request added https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/3200
